### ISSUE TITLE
Compilation problems using IAR 

### ISSUE THREAD

**Issue by EdertTadeu** [aws/aws-iot-device-sdk-embedded-C#204]
I downloaded en.x-cube-aws package from ST website and it uses aws-iot-device-sdk-embedded-C v2.0, so I decided update to v3.0. Compiling the IAR project with v3.0, the compiler complains the follow issues:

File "aws_timer.c":
=> No function prototypes.

File "aws_iot_mqtt_client.c"
=> No function prototype for "aws_iot_mqtt_set_client_state( )"

File "aws_iot_mqtt_client_connect.c"
=> No function prototype for "_aws_iot_mqtt_internal_disconnect( )"

Solving this problems, the IAR project will compile.

**Comment by gordonwang0** [comment]
Hi @EdertTadeu,

There is no file called aws_timer.c in this SDK.  Did you mean timer.c?

The function prototype for `aws_iot_mqtt_set_client_state` is in `aws_iot_mqtt_client_common_internal.h`.
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/d039f075e1cc2a2a7fc20edc6868f328d8d36b2f/include/aws_iot_mqtt_client_common_internal.h#L119

The prototype for _aws_iot_mqtt_internal_disconnect does seem to be missing.

Please allow me me take a few days to look into these issues.  But please keep in mind that we can give general support and guidance on this SDK, and not support for specific hardware or compilers.

**Comment by gordonwang0** [comment]
Hi @EdertTadeu,

#205 will fix the function prototypes issue with `aws_iot_mqtt_set_client_state( )` and `_aws_iot_mqtt_internal_disconnect( )`.  The file `aws_timer.c` is not a part of this SDK (it appears to be something that ST implemented).  You will have to add the prototypes for those functions yourself.

**Comment by gordonwang0** [comment]
Hi @EdertTadeu,

Please let me know if you need any more help with this.  Closing issue.
